measure.rb

```diff
@@ -3028,6 +3029,43 @@ def self.add_output_variables(model, vars, objects)
     end
   end
 
+  def self.add_ems_cooling_heating_load_output(model)
+    control_zones = @control_slave_zones_hash.keys
+    control_zones.each do |living_zone|
+      # sensors
+      load_rate_sensor = OpenStudio::Model::EnergyManagementSystemSensor.new(model, "Zone Predicted Sensible Load to Setpoint Heat Transfer Rate")
+      load_rate_sensor.setName("#{living_zone.name} Sensible Load Rate")
+      load_rate_sensor.setKeyName(living_zone.name.to_s)
+
+      # program
+      load_program = OpenStudio::Model::EnergyManagementSystemProgram.new(model)
+      load_program.setName("#{living_zone.name} clg htg load output program")
+      load_program.addLine("Set #{living_zone.name}_htg_load = 0")
+      load_program.addLine("Set #{living_zone.name}_clg_load = 0")
+      load_program.addLine("If #{load_rate_sensor.name} > 0")
+      load_program.addLine("Set #{living_zone.name}_htg_load = #{load_rate_sensor.name} * 3600")
+      load_program.addLine("Else")
+      load_program.addLine("Set #{living_zone.name}_clg_load = - #{load_rate_sensor.name} * 3600")
+      load_program.addLine("EndIf")
+
+      # ems output variables
+      ['clg', 'htg'].each do |load_type|
+        @hvac_map["BLDGLOAD:#{living_zone.name}_#{load_type}_load"] = []
+        ems_output_load = OpenStudio::Model::EnergyManagementSystemOutputVariable.new(model, "#{living_zone.name}_#{load_type}_load")
+        ems_output_load.setName("#{living_zone.name} #{load_type} load")
+        ems_output_load.setTypeOfDataInVariable("Summed")
+        ems_output_load.setUpdateFrequency("ZoneTimestep")
+        ems_output_load.setEMSProgramOrSubroutineName(load_program)
+        ems_output_load.setUnits("J")
+        @hvac_map["BLDGLOAD:#{living_zone.name}_#{load_type}_load"] << ems_output_load
```

Comment:
@shorowit Since loads are parsed by building unit not system unit, which is different from other mapping data here, I'm not sure if this is the best place to store EMS load output names.